packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -55,22 +55,34 @@ MenuWrapper.propTypes = {
 };
 
 /**
- * Extracts all focusable children from an html tree.
+ * Extracts all focusable children from an html tree, optionally ignoring items from submenu.
  *
  * @param {HTMLElement} parent The parent to search
+ * @param {boolean} isSubMenu If we're searching from submenu.
  * @return {Array.<HTMLElement>} List of focusable elements
  */
-function getFocusableChildren(parent) {
-  return Array.from(parent.querySelectorAll(FOCUSABLE_SELECTORS.join(', ')));
+function getFocusableChildren(parent, isSubMenu) {
+  const allButtons = Array.from(
+    parent.querySelectorAll(FOCUSABLE_SELECTORS.join(', '))
+  );
+  if (isSubMenu) {
+    return allButtons;
+  }
+  // Skip considering the submenu, and the submenu items.
+  return allButtons.filter(
+    (elem) => !elem.matches('[role="menu"], [role="menu"] [role="menu"] *')
+  );
```

Comment:
Could this not be added permanently to the `FOCUSABLE_SELECTORS` - that they should be something like `button:not(:disabled):not([role=menu])`? Or maybe it's the last entry, the tab index one, that also includes the menu?

If not, alternatively make two different arrays of selectors, and just use the proper one here:

```
function getFocusableChildren(parent) {
  const selectors = isSubMenu ? FOCUSABLE_SUBMENU_SELECTORS : FOCUSABLE_SELECTORS;
  return Array.from(parent.querySelectorAll(selectors.join(', ')));
```

Reply:
I tried out different selectors for being able to ignore items that are children of menu but didn't find a selector, thus opted for filtering out the submenu items if it's not the submenu.

The problem is that `button:not(:disabled)` inside the parent will also select all the non-disabled buttons of the submenu inside that menu. Even if ignoring the submenu itself as an item is doable, couldn't find a simple way to ignore the items inside the submenu. Anyway, I can try again.

Reply:
OK, so I tried a bunch of different options again creating a separate selector array for the parent menu, e.g. `:not([role="menu"]) button...` and `...:not([role="menu"])` etc. but these didn't work, still the submenu items are getting included. Let me know if you know a selector which would exclude children of a determined parent ([role="menu"] in this case).

Reply:
> Let me know if you know a selector which would exclude children of a determined parent (`[role="menu"]` in this case).

Ah no, you can't do that. Maybe with `:has()`, but that isn't supported yet.

It wasn't a big deal, just looked like something that could potentially be optimized a bit. But if not, fine with me!